updater/update/update.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2021 PaperCut Software International Pty. Ltd.
+ */
+
+package update
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type UpgradeInfo struct {
+	URL        string
+	Version    string
+	Sha1       string
+	Sha256     string
+	Operations []Operation
+}
+
+type Operation struct {
+	Action string
+	Args   []string
+}
+
+func Check(url string, currentVer string) (*UpgradeInfo, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", url+"?version="+currentVer, nil)
```

Comment:
maybe encode query param the currentVer to be on the safe?

Reply:
This is unchanged really to how it was for all these years, but suppose can do with extra safety

Reply:
done